File: Typscript/Step-1 Basic/Maths/reverseNum.ts
// Time Complexity: O(Log10(n))

function reverse(n:number): number {
    let reverse = 0;
    while(n > 0) {
        let lastDigit = n%10
        reverse = reverse*10 + lastDigit
        n = Math.floor(n/10)
    }
    return reverse;
}

console.log(reverse(1234))

File: Go/Step1/Maths/reverseNum.go
package main

// Time Complexity O(log10N)

func reverseNum(num int) int {
	rev := 0

	for num > 0 {
		lastDigit := num % 10
		rev = rev*10 + lastDigit
		num = num / 10
	}
	return rev
}
